crates/albedo_rtx/src/lib.rs

```diff
@@ -14,3 +16,210 @@ pub fn get_dispatch_size(
+                sample_type,
+                view_dimension,
+            },
+            count: None,
+        }
+    }
+
+    pub fn sampler(binding: u32, visibility: wgpu::ShaderStages, binding_type: wgpu::SamplerBindingType) -> wgpu::BindGroupLayoutEntry {
+        wgpu::BindGroupLayoutEntry {
+            binding,
+            visibility,
+            ty: wgpu::BindingType::Sampler(binding_type),
+            count: None,
+        }
+    }
+
+    pub fn create(device: &wgpu::Device) -> Self {
+        let layout = device.create_bind_group_layout(&wgpu::BindGroupLayoutDescriptor {
+            label: Some("Scene Layout"),
+            entries: &[
+                wgpu::BindGroupLayoutEntry {
+                    binding: Self::PER_DRAW_STRUCT_BINDING,
+                    visibility: wgpu::ShaderStages::COMPUTE,
+                    ty: wgpu::BindingType::Buffer {
+                        ty: wgpu::BufferBindingType::Uniform,
+                        has_dynamic_offset: false,
+                        min_binding_size: None,
+                    },
+                    count: None,
+                },
+                Self::buffer(Self::INSTANCE_BINDING, wgpu::ShaderStages::COMPUTE,
+                    true),
+                Self::buffer(Self::NODE_BINDING, wgpu::ShaderStages::COMPUTE,
+                    true),
+                Self::buffer(Self::INDEX_BINDING, wgpu::ShaderStages::COMPUTE,
+                    true),
+                Self::buffer(Self::VERTEX_BINDING, wgpu::ShaderStages::COMPUTE,
+                    true),
+                Self::buffer(Self::LIGHT_BINDING, wgpu::ShaderStages::COMPUTE,
+                    true),
+                Self::buffer(Self::MATERIAL_BINDING, wgpu::ShaderStages::COMPUTE,
+                    true),
+                Self::texture(Self::TEXTURE_PROBE_BINDING, wgpu::ShaderStages::COMPUTE,
+                    wgpu::TextureSampleType::Float { filterable: true }, wgpu::TextureViewDimension::D2),
+                Self::texture(Self::TEXTURE_INFO_BINDING, wgpu::ShaderStages::COMPUTE,
+                    wgpu::TextureSampleType::Uint, wgpu::TextureViewDimension::D1),
+                Self::texture(Self::TEXTURE_ATLAS_BINDING, wgpu::ShaderStages::COMPUTE,
+                    wgpu::TextureSampleType::Float { filterable: true }, wgpu::TextureViewDimension::D2Array),
+                Self::sampler(Self::SAMPLER_BINDING, wgpu::ShaderStages::COMPUTE,
+                    wgpu::SamplerBindingType::Filtering),
+                Self::sampler(Self::SAMPLER_LINEAR_BINDING, wgpu::ShaderStages::COMPUTE,
+                    wgpu::SamplerBindingType::Filtering),
+            ],
+        });
+        Self { layout }
+    }
+    
+    pub fn bind_group(&self, bindings_count: u32) -> SceneBindGroup {
+        SceneBindGroup{
+            layout: &self.layout,
+            bindings: Vec::<wgpu::BindGroupEntry>::with_capacity(bindings_count as usize)
+        }
+    }
+}
+
+impl<'a> SceneBindGroup<'a> {
```

Comment:
I would refactor with a BindGroupBuilder as well, and create overloads like `sub_buffer` etc... To allow setting the max size or offset. I already used the max size on the main branch